src/OpenTracing/ITracer.cs

```diff
@@ -1,33 +1,112 @@
+        ///         A contrived example:
+        ///         <code>
+        /// ITracer tracer = ...
+        /// 
+        /// // Note if there is a ITracer.ActiveSpan(), it will be used as the target of an implicit CHILD_OF
+        /// // Refernece for "workSpan" when StartActive() is invoked.
+        /// using (IActiveSpan workSpan = tracer.BuildSpan("DoWork").StartActive())
+        /// {
+        ///     workSpan.SetTag("...", "...");
+        ///     // etc, etc
+        /// }
+        /// 
+        /// // It's also possible to create Spans manually, bypassing the ActiveSpanSource activation.
+        /// Span http = tracer.BuildSpan("HandleHTTPRequest")
+        ///                   .AsChildOf(rpcSpanContext)  // an explicit parent
+        ///                   .WithTag("user_agent", req.UserAgent)
+        ///                   .WithTag("lucky_number", 42)
+        ///                   .StartManual();
+        ///         </code>
+        ///     </para>
         /// </summary>
-        /// <param name="operationName">The operation name of the span.</param>
+        /// <param name="operationName"></param>
         ISpanBuilder BuildSpan(string operationName);
 
         /// <summary>
-        /// Inject a <see cref="ISpanContext"/> into a <paramref name="carrier"/> of a given type,
-        /// presumably for propagation across process boundaries.
+        ///     Inect a SpanContext into a 'carrier' of a given type, presumably for propagation across process boundaries.
+        ///     <para>
+        ///         Example:
+        ///         <code>
+        /// Tracer tracer = ...
+        /// Span clientSpan = ...
+        /// TextMap httpHeadersCarrier = new AnHttpHeaderCarrier(httpRequest);
+        /// tracer.Inject(span.Context(), BuiltinFormats.HttpHeaders, httpHeadersCarrier);
+        /// </code>
+        ///     </para>
         /// </summary>
-        /// <param name="spanContext">The <see cref="ISpanContext"/> instance to inject into the carrier.</param>
-        /// <param name="format">The format of the carrier.</param>
-        /// <param name="carrier">See the documentation for the chosen <paramref name="format"/> for a description of the carrier object.</param>
-        void Inject<TCarrier>(ISpanContext spanContext, Format<TCarrier> format, TCarrier carrier);
+        /// <typeparam name="T">The carrier type, which also parametrizes the Format.</typeparam>
+        /// <param name="spanContext">The SpanContext instance to inject into the carrier</param>
+        /// <param name="format">The Fromat of the carrier</param>
+        /// <param name="carrier">
+        ///     The carrier for the SpanContext state. All Tracer.Inject implementations must support
+        ///     <see cref="TextMap" /> and <see cref="Stream" />
+        /// </param>
+        /// <seealso cref="IFormat{T}" />
+        /// <seealso cref="BuiltinFormats" />
+        void Inject<T>(ISpanContext spanContext, IFormat<T> format, T carrier);
```

Comment:
It's not common in .NET to use just "T" - why drop a more explicit "TCarrier"?

Reply:
resolved